go/ct/st/transaction_context_test.go

```diff
@@ -43,16 +47,78 @@ func TestTransactionContext_String(t *testing.T) {
 			t.OriginAddress[19] = 0xfe
 			return t.OriginAddress
 		}},
+		"Blob Hashes": {func(t *TransactionContext) any {
+			t.BlobHashes = []vm.Hash{{1, 2, 3, 4}}
+			return t.BlobHashes
+		}},
 	}
 
-	c := TransactionContext{}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			c := TransactionContext{}
 			v := test.change(&c)
-			str := c.String()
-			if !strings.Contains(str, fmt.Sprintf("%v: %v", name, v)) {
+			got := c.String()
+			want := fmt.Sprintf("%v: %v", name, v)
+			if !strings.Contains(got, want) {
 				t.Errorf("Did not find %v string", name)
 			}
 		})
 	}
 }
+
+func TestTransactionContext_Clone(t *testing.T) {
+
+	baseAddress := vm.Address{1, 2, 3, 4}
+	baseHash := vm.Hash{1, 2, 3, 4}
+
+	t1 := TransactionContext{
+		OriginAddress: baseAddress,
+		BlobHashes:    []vm.Hash{baseHash},
+	}
+
+	t2 := t1.Clone()
+
+	if !reflect.DeepEqual(t1.BlobHashes, t2.BlobHashes) || !reflect.DeepEqual(t1.OriginAddress, t2.OriginAddress) {
+		t.Errorf("Cloned transaction context is not equal to original")
+	}
+
+	t2.OriginAddress[0] = 0xff
+	t2.BlobHashes[0] = vm.Hash{0x00}
+
+	if reflect.DeepEqual(t1.BlobHashes, t2.BlobHashes) || reflect.DeepEqual(t1.OriginAddress, t2.OriginAddress) {
+		t.Errorf("Cloned transaction context is not independent to original")
+	}
+}
+
+func TestTransactionContext_Equal(t *testing.T) {
+
+	tests := map[string]struct {
+		change func(*TransactionContext, *TransactionContext)
+		wanted bool
+	}{
+		"match": {func(t1, t2 *TransactionContext) {
+			t1.BlobHashes = []vm.Hash{{1}}
+			t2.BlobHashes = []vm.Hash{{1}}
+		}, true},
+		"mismatch": {func(t1, t2 *TransactionContext) {
+			t1.BlobHashes = []vm.Hash{{1}}
+			t2.BlobHashes = []vm.Hash{{4}}
+		}, false},
+		"mismatch-different-blobhash-length": {func(t1, t2 *TransactionContext) {
+			t1.BlobHashes = []vm.Hash{{1}, {2}}
+			t2.BlobHashes = []vm.Hash{{4}}
+		}, false},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			t1 := TransactionContext{}
+			t2 := TransactionContext{}
+			test.change(&t1, &t2)
```

Comment:
You could add an equals check before applying the change, this would help with debugging if the Equals method is broken.